### ISSUE TITLE
Thanks!

### ISSUE THREAD

**Issue by xied75** [YallaDotNet/SyslogNet#1]
Just curious, how did you convert the upstream into portable libs? I need to do something similar and wonder what procedure/best practices you got during your work?

Best regards,

Dong

**Comment by dmitry-shechtman** [comment]
I don't think there is a procedure. I can only describe what I did with this library.

Porting the senders was super-easy. I just used sockets-for-PCL, which had a mostly compatible (apart from the async bit) API.

The biggest problem was ASCIIEncoding, which I was tempted to leave as a parameter, but ultimately repackaged a small subset of Portable.Text.Encoding.

One of my goals was retaining maximum compatibility with upstream, which is why I ended up releasing three packages (with five assemblies between them). If you don't have such concerns, your job will probably be a lot easier.

Bottom line: there is a free portable library for nearly anything nowadays, just search, and remember to leave the copyright and license intact ☺

Good luck with your porting effort!
